kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/ApplicationCommandRegistry.kt

```diff
@@ -72,324 +72,114 @@ public open class ApplicationCommandRegistry : KoinComponent {
         }
 
     /** Handles the initial registration of commands, after extensions have been loaded. **/
-    public open suspend fun initialRegistration() {
+    public suspend fun initialRegistration() {
         if (initialised) {
             return
         }
-
-        if (!bot.settings.applicationCommandsBuilder.register) {
-            logger.debug {
-                "Application command registration is disabled, pairing existing commands with extension commands"
-            }
-        }
-
         val commands: MutableList<ApplicationCommand<*>> = mutableListOf()
-
         bot.extensions.values.forEach {
             commands += it.messageCommands
             commands += it.slashCommands
             commands += it.userCommands
         }
-
-        try {
-            syncAll(true, commands)
-        } catch (t: Throwable) {
-            logger.error(t) { "Failed to synchronise application commands" }
-        }
-
+        initialize(commands)
```

Comment:
Should probably wrap this with a `try` block to make sure `initialised` is always set.